lib/std/special/docs/index.html

```diff
@@ -5,62 +5,228 @@
-        text-decoration: underline;
-        margin: 0;
-        padding: 0.5em 0;
-        text-align: center;
+
+      .flex-right {
+        display: flex;
+        overflow: auto;
+        flex-grow: 1;
+        flex-shrink: 1;
+        
+        z-index: 200;
+      }
+
+      .flex-right > .wrap {
+        width: 60rem;
+        max-width: 85vw;
+        flex-shrink: 1;
+      }
+      
+      .help-modal {
+        z-index: 400;
+      }
+      
+      /* sidebar */
+      .sidebar {
+        font-size: 1rem;
+        background-color: var(--bg-color);
+        box-shadow: 0 0 1rem var(--sidebar-sh-color);
+      }
+      
+      .sidebar .logo {
+        padding: 1rem 0.35rem 0.35rem 0.35rem;
+      }
+      
+      .sidebar .logo > svg {
+        display: block;
+        overflow: visible;
+      }
+      
+      .sidebar h2 {
+        margin: 0.5rem;
+        padding: 0;
+        font-size: 1.2rem;
       }
-      nav p {
+      
+      .sidebar h2 > span {
+        border-bottom: 0.125rem dotted var(--tx-color);
+      }
+      
+      .sidebar .packages {
+        list-style-type: none;
         margin: 0;
         padding: 0;
-        text-align: center;
+        background-color: var(--sidebar-pkg-bg-color);
+      }
+      
+      .sidebar .packages > li > a {
+        display: block;
+        padding: 0.5rem 1rem;
+        color: var(--sidebar-pkglnk-tx-color);
+        background-color: var(--sidebar-pkglnk-bg-color);
+        text-decoration: none;
+      }
+      
+      .sidebar .packages > li > a:hover {
+        color: var(--sidebar-pkglnk-tx-color-hover);
+        background-color: var(--sidebar-pkglnk-bg-color-hover);
+      }
+      
+      .sidebar .packages > li > a.active {
+        color: var(--sidebar-pkglnk-tx-color-active);
+        background-color: var(--sidebar-pkglnk-bg-color-active);
+      }
+      
+      .sidebar p.str {
+        margin: 0.5rem;
+        font-family: var(--mono);
+      }
+
+      /* docs section */
+      .docs {
+        padding: 1rem 0.7rem 2.4rem 1.4rem;
+        font-size: 1rem;
+        background-color: var(--bg-color);
+        overflow-wrap: break-word;
+      }
+
+      .docs .search {
+        width: 100%;
+        margin-bottom: 0.8rem;
+        padding: 0.5rem;
+        font-size: 1rem;
+        font-family: var(--ui);
+        color: var(--tx-color);
+        background-color: var(--search-bg-color);
+        border-top: 0;
+        border-left: 0;
+        border-right: 0;
+        border-bottom-width: 0.125rem;
+        border-bottom-style: solid;
+        border-bottom-color: var(--tx-color);
+        outline: none;
+        transition: border-bottom-color 0.35s, background 0.35s, box-shadow 0.35s;
+      }
+      
+      .docs .search:focus {
+        background-color: var(--search-bg-color-focus);
+        border-bottom-color: #ffbb4d;
+        box-shadow: 0 0.3em 1em 0.125em var(--search-sh-color);
+      }
+      
+      .docs .search::placeholder {
```

Comment:
same question regarding `::`

Reply:
Pseudo-element selector that allows to apply styles to `<input>`'s `placeholder` attribute (its not affected by other means and does not inherit from `<input>`).

Chrome 57+, Firefox 51+, Safari 10+.
(Should be silently ignored if not supported)